src/components/LibraryStats.tsx

```diff
@@ -1,180 +1,294 @@
-                    angle={-45}
-                    textAnchor="end"
-                    padding={{ left: 50, right: 50 }}
-                    height={175}
-                  />
-                  <YAxis hide={true} />
-                  <Tooltip />
-                  <Bar
-                    stackId="collections"
-                    dataKey="Licensed Titles"
-                    barSize={50}
-                    fill="#737373"
-                  />
-                  <Bar
-                    stackId="collections"
-                    dataKey="Open Access Titles"
-                    barSize={50}
-                    fill="#000000"
-                  />
-                </BarChart>
-              </li>
-            )}
-          </ul>
-        )}
-      </div>
-    );
-  }
-
-  formatNumber(n) {
-    return n ? numeral(n).format("0.[0]a") : 0;
-  }
-}
+ for a single library or all libraries the admin has access to. */
+const LibraryStats = (props: LibraryStatsProps) => {
+  const {
+    stats: {
+      name: libraryName,
+      key: libraryKey,
+      collections,
+      inventorySummary: inventory,
+      patronStatistics: patrons,
+    },
+    library,
+  } = props;
+
+  const chartItems =
+    props.stats &&
+    props.stats.collections &&
+    collections
+      .map(({ name, inventory }) => ({ name, ...inventory }))
+      .sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1));
+
+  /* Customize the Rechart tooltip to provide additional information */
+  const CustomTooltip = (props) => {
+    const { active, payload } = props;
+    if (!active) return null;
+
+    // Nab inventory data from one of the chart payload objects.
+    const chartInventory = payload[0].payload;
+    const aboveTheLineColor = "#030303";
+    const belowTheLineColor = "#A0A0A0";
+    const aboveTheLine: chartTooltipData[] = [
+      {
+        dataKey: "titles",
+        name: inventoryKeyToLabelMap.titles,
+        value: chartInventory.titles,
+      },
+      {
+        dataKey: "availableTitles",
+        name: inventoryKeyToLabelMap.availableTitles,
+        value: chartInventory.availableTitles,
+      },
+      ...payload.filter(({ value }) => value > 0),
+    ].map((entry) => ({ ...entry, color: aboveTheLineColor }));
+    const aboveTheLineKeys = [
+      "name",
+      ...aboveTheLine.map(({ dataKey }) => dataKey),
+    ];
+    const belowTheLine = Object.entries(chartInventory)
+      .filter(([key]) => !aboveTheLineKeys.includes(key))
+      .map(([key, value]) => ({
+        dataKey: key,
+        name: inventoryKeyToLabelMap[key],
+        value,
+        color: belowTheLineColor,
+      }));
+    const newPayload = [
+      ...aboveTheLine,
+      {}, // blank line
+      { value: ">>> Additional Information <<<", color: belowTheLineColor },
+      ...belowTheLine,
+    ];
+
+    // We render the default, but with our overridden payload.
+    return <DefaultTooltipContent {...props} payload={newPayload} />;
+  };
```

Comment:
This bit overrides the tooltip to display inventory details for each of the individual collections in the bar chart. It's kind of messy, so would especially appreciate any suggestions for cleaning it up.

Reply:
I'm overriding only the `payload` here, so I'll need the rest of the `props` to pass on to the default tooltip. So, I'll keep this as a React component and reference it properly on the chart `<Tooltip />`.

Reply:
I was previously referencing something inside the closure, which is why it was there; but that is no longer the case.